ractor_cluster/src/net/listener.rs

```diff
@@ -65,15 +65,36 @@ impl Actor for Listener {
             }
         };
 
-        // startup the event processing loop by sending an initial msg
-        let _ = myself.cast(ListenerMessage);
-
         // create the initial state
         Ok(Self::State {
             listener: Some(listener),
         })
     }
 
+    async fn post_start(
```

Comment:
is there a reason why you don't want to do this just directly in post-start? 

Instead of using the untyped supervisor, I'd prefer passing a typed `ActorRef<TSupervisorMessage>` in the arguments to the listener, the listener opens the port, and replies back to the supervisor. Then you can just move on with the traditional startup.

Reply:
The reason I added the `post_start` was that the `try_get_supervisor()` returned `None` in the `pre_start`.
With passing a ActorRef on startup thats not needed of course.

You said `ActorRef<TSupervisorMessage>` - am I right doing this with `ActorRef<NodeServerMessage>`? I couldn't find a `TSupervisorMessage` and a `SupervisionEvent` wouldn't be suitable?

Reply:
Yeah sorry I was on my phone so couldn't find the type quickly and was trying to use a generic value. This looks good to me, thanks!